src/envoy/http/authn/http_filter.cc

```diff
@@ -74,6 +74,12 @@ void AuthenticationFilter::onPeerAuthenticationDone(bool success) {
 }
 void AuthenticationFilter::onOriginAuthenticationDone(bool success) {
   ENVOY_LOG(debug, "{}: success = {}", __func__, success);
+  // After Istio authn, the JWT headers consumed by Istio authn should be
```

Comment:
Not always true, if peer authn fail, it will not reach origin auth part. Maybe you should put this in onDestroy (or alike) function.

Reply:
If peer authn fails,  rejectRequest("Peer authentication failed.") is called, the http client receives a 401 code, and the request will not proceed.